### PR TITLE
Skip session data recording for networking requests

### LINKED ISSUE #48728: Extra session data entry with no sid
Expected behavior:

Only one session data record with the session id is recorded. 

Current behavior:

When a ssh session is exited on macos a extra session data entry is done.  No session id is associated with the record. This is not reproducible on Linux or on earlier Teleport versions.

Image

Bug details:
- Teleport version: 17.0.0 beta 4
- Recreation steps:

1. Install v17 SSH service on macos
2. Run a ssh session and exit
3. Review audit log entries

### PR DESCRIPTION
**Pull request by Joerger**
The new networking process for v17 inherited a small bug in the forwarding server context which causes a `session.data` audit event with missing fields. This PR skips the audit event by setting recording mode to off, but a better long term solution would be to [separate "connection", "command", and "server" contexts](https://github.com/gravitational/teleport/pull/45562). `ServerContext` is just trying to do too much and isn't very reusable.

Fixes https://github.com/gravitational/teleport/issues/48728

### CODE REVIEW

**Review by marcoandredinis — APPROVED** [review]

File: lib/srv/regular/sshserver.go [hunk 1]
```diff
@@ -1180,6 +1180,7 @@ func (s *Server) startNetworkingProcess(scx *srv.ServerContext) (*networking.Pro
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
+	nsctx.SessionRecordingConfig.SetMode(types.RecordOff)
```
Comment (codingllama) [inline lib/srv/regular/sshserver.go:1183]: Could we add a test for the bug scenario so we don't hit it again?

File: lib/srv/regular/sshserver_test.go [hunk 2]
```diff
@@ -443,6 +443,112 @@ loop:
 	require.Equal(t, "echo 1", execEvent.CommandMetadata.Command)
 }
 
+// TestSessionAuditLog tests that the expected audit events are emitted for a
+// session with various subsystems involved.
+//
+// Note: This is a regression test for a bug which resulted in extra, empty session.data
+// events for networking requests.
```
Comment (codingllama) [inline lib/srv/regular/sshserver_test.go:450]:
```suggestion
// events for networking requests.
// See https://github.com/gravitational/teleport/issues/48728.
```

File: lib/srv/regular/sshserver_test.go [hunk 3]
```diff
@@ -443,6 +443,112 @@ loop:
...
+	}
+
+	timeout := time.After(time.Second)
+	var gotEvents []apievents.AuditEvent
+loop:
+	for {
+		select {
+		case event := <-emitter.C():
+			fmt.Printf("type %T\n", event)
+			gotEvents = append(gotEvents, event)
```
Comment (codingllama) [inline lib/srv/regular/sshserver_test.go:537]: Should we stop once len(gotEvents) matches len(expectedEvents)?

File: lib/srv/regular/sshserver_test.go [hunk 4]
```diff
@@ -443,6 +443,112 @@ loop:
...
+			break loop
+		}
+	}
+
+	if len(expectedEvents) != len(gotEvents) {
+		require.Fail(t, "collected audit events don't match expected events", "got %v, expected %v", gotEvents, expectedEvents)
+	}
+
+	for i, ee := range expectedEvents {
+		require.IsType(t, ee, gotEvents[i])
```
Comment (codingllama) [inline lib/srv/regular/sshserver_test.go:548]: Should we check that the session ID isn't empty?

**Review by codingllama — APPROVED** [review]

**Pull request merged**